### PR TITLE
fix(cli): reject mixed staged configuration snapshots

### PR DESCRIPTION
**Pull request by aidenybai**
## Summary

- refuse `--staged` scans when known governing configuration differs between the Git index and worktree
- preserve raw porcelain status columns so unstaged, untracked, ignored, nested, and ancestor configuration changes are detected
- emit an anonymized divergence counter and return a schema-valid staged-mode error report

## Why

Staged source was materialized from the index while `doctor.config.json` was loaded from the worktree. The same indexed source could therefore gain a false positive or lose a true diagnostic solely because of an unstaged config edit. A staged verdict must use one snapshot. Until all executable configuration and plugin inputs can be materialized from the index, rejecting a mixed snapshot is the conservative contract.

## Validation

- exact 12-run published/current matrix from `/tmp/rd-v12-staged-config-snapshot.mjs`
  - aligned error/error still reports `button-has-type` and exits 1
  - aligned off/off stays clean and exits 0
  - both crossed snapshots now return a staged-mode `CliInputError`, zero diagnostics, and exit 1
- 7 focused Git regression cases covering ordinary staged source edits, aligned staged config, unstaged tracked config, nested config, ancestor config, untracked/ignored config, and non-Git input
- `nr test` (CLI 2,215 passed; core 1,286 passed; plugin, API, language-server, ESLint mirror, and fuzz suites passed)
- `nr typecheck`
- `nr lint`
- `nr format:check`
- `nr smoke:json-report`
- `git diff --check`
- post-change `truffler` dedup search

## Compatibility

- patch changeset for `react-doctor`
- JSON schema remains version 3
- no GitHub Action release files changed

## Product boundary

The rollback signal is the `staged.snapshot_divergence` rejection rate. If legitimate partial-staging workflows hit this too often, the replacement is full index-snapshot materialization, not a return to mixed-snapshot verdicts.

---

> [!NOTE]
> **Medium Risk**
> Changes pre-commit and `--staged` behavior for repos with partially staged or unstaged config; legitimate workflows may need to stage or restore config before scanning.
> 
> **Overview**
> **`--staged` scans now fail fast** when governing config on disk does not match what Git has indexed, so staged source from the index is not judged against worktree-only config (and the opposite).
> 
> Before materializing staged files, `inspect` runs `findStagedSnapshotDivergences` on the **resolved** scan root (after `rootDir` redirect). Git porcelain status (`-z`, untracked/ignored) is parsed for known config basenames—`doctor.config.*`, ESLint/Vite/Next/Babel tooling configs, plus core staged materialization names—and any worktree/index mismatch yields a `CliInputError` listing paths. Non-Git or unverifiable environments get a separate error; hits record `staged.snapshot_divergence`. JSON mode still emits a schema-valid **staged** error report.
> 
> `runGit` is split so **`runGitRaw`** preserves full git stdout for NUL-delimited parsing. Regression tests cover parser edge cases, integration repos, and the `inspectAction` guard (including `--json` and redirected `rootDir`).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f2edb7a1cdfd691d540bc2814b98e9cde1f4b0fc. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/react-doctor/src/cli/utils/find-staged-snapshot-divergences.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  const divergentConfigFiles = new Set<string>();
+  const statusEntries = statusOutput.split("\0").filter((entry) => entry.length > 0);
+  for (let entryIndex = 0; entryIndex < statusEntries.length; entryIndex += 1) {
+    const entry = statusEntries[entryIndex];
+    const [indexStatus, worktreeStatus] = entry;
+    const filePath = entry.slice("XY ".length);
+    if (worktreeStatus !== " " && SNAPSHOT_CONFIG_FILENAMES.has(path.basename(filePath))) {
+      divergentConfigFiles.add(filePath);
+    }
+    if (indexStatus === "R" || indexStatus === "C") entryIndex += 1;
```
[the remark this answers is not in the document]
Comment (aidenybai) [reply packages/react-doctor/src/cli/utils/find-staged-snapshot-divergences.ts]: Fixed in 7555454f0. The parser now consumes rename/copy source records when either porcelain status column is R/C and checks both worktree-side paths for governing config names. Added synthetic worktree rename/copy regressions plus an aligned index-only rename control; focused tests pass 10/10 and package typecheck passes.

**Review by rayhanadev — APPROVED** [review]
Adversarial review: 3 proven + 1 confirmed findings, all fixed in 748e6d2df/f2edb7a1c.
- Divergence set no longer unions the cache-fingerprint list — unstaged lockfile/.gitignore/pnpm-workspace.yaml edits no longer hard-block --staged (the pre-commit-hook FP).
- Staged guard now runs against the rootDir-resolved directory, closing the redirect bypass.
- inspectAction staged wiring pinned (divergence rejection, non-git, --json mode "staged", redirect); vacuous index-only-rename test replaced with mutant-killing cases.
Deslop −3 added lines (executeGit indirection collapsed). react-doctor suite 2225 passed; typecheck/lint/format clean.
Follow-ups (not blocking): skip trailing-slash "!!" porcelain entries if the parser is touched again (ignored-dir-named-like-a-config FP, contrived reachability); migrate scan-result-cache's parseWorktreeStatusRecords onto runGitRaw and drop its trim HACK.
